src/utils/dateUtils/dateCalculations.test.js

```diff
@@ -0,0 +1,319 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import {
+  calculateDateDaysInPast,
+  calculateDateDaysInPastFromToday,
+  calculateDateMonthsInPast
+} from './dateCalculations';
+
+import * as dateCalculations from './dateCalculations';
+
+const YEAR = 2018;
+const PREVIOUS_YEAR = 2017;
+
+const JANUARY = 0;
+const FEBRUARY = 1;
+const APRIL = 3;
+const DECEMBER = 11;
+
+describe('calculateDateDaysInPast', () => {
+  it('should properly calculate days in past', () => {
+    expect(calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), 5))
+      .to
+      .deep
+      .equal(new Date(YEAR, JANUARY, 10));
+  });
+  it('should properly calculate given a string for days', () => {
+    expect(calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), '5'))
+      .to
+      .deep
+      .equal(new Date(YEAR, JANUARY, 10));
+  });
+  it('should throw an error if days is an object', () => {
+    expect(() => {
+      calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), {});
+    }).to
+      .throw();
+  });
+  it('should throw an error if days is a function', () => {
+    expect(() => {
+      calculateDateDaysInPast(
+        new Date(YEAR, JANUARY, 15),
+        () => console.log('a function')
+      );
+    }).to
+      .throw();
+  });
+  it('should throw an error if days is null', () => {
+    expect(() => {
+      calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), null);
+    }).to
+      .throw();
+  });
+  it('should throw an error if days is an array', () => {
+    expect(() => {
+      calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), []);
+    }).to
+      .throw();
+  });
+  it('should throw an error if days is a boolean', () => {
+    expect(() => {
+      calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), true);
+    }).to
+      .throw();
+  });
+  it('should throw an error if days is a non-numeric string', () => {
+    expect(() => {
+      calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), 'hello');
+    }).to
+      .throw();
+  });
+  it('should throw an error if days is letters followed by numbers', () => {
+    expect(() => {
+      calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), 'h15');
+    }).to
+      .throw();
+  });
+  it('should throw an error if days is numbers followed by letters', () => {
+    expect(() => {
+      calculateDateDaysInPast(new Date(YEAR, JANUARY, 15), '15h');
+    }).to
+      .throw();
+  });
+  it('should throw an error if date is an object', () => {
```

Comment:
A case where calculateDateDaysInPast can be tricked is if you pass it an object like this as `date`:
```javascript
const myTrickyDate = { getDate: function() { console.log('test') }, setDate: function() { console.log('test') } }

calculateDateDaysInPast(myTrickyDate, 5); // No exception thrown
```
There are some suggestions here for how to deal with this: https://stackoverflow.com/q/643782.  A robust solution is harder to come by than I expected.